File: SiHyeon/week2/멀리 뛰기.cpp
#include <string>
#include <array>

using namespace std;
array<long long, 9999> memory = { 0, };

long long fibonacci(long long n) {
    long long result;
    if (n <= 1) return n;
    long long index = n;
    if (memory[index])
        return memory[index];
    else
        result = (memory[index] = fibonacci(n - 1) + fibonacci(n - 2));
    return result % 1234567;
}

long long solution(int n) {
    long long answer = fibonacci(n + 1);
    return answer % 1234567;
}

File: yejee/week2/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        long answer = 0;
        long[] arr = new long[n + 1]; //배열의 크기가 1인 경우 7번줄의 실행을 위해 +1
        
        arr[0] = 1;
        arr[1] = 2;
        
        for(int i = 2 ; i < n ; i++){ //피보나치
            arr[i] = (arr[i - 2] + arr[i - 1]) % 1234567;
        }
        
        answer = arr[n - 1];
        
        return answer;
    }
}